packages/cinterop/src/jvm/jni/utils.cpp

```diff
@@ -82,13 +82,12 @@ static std::string string_to_hex(const std::string& message, realm::StringData&
     ret << "error_code = " << error_code << "; ";
     ret << "retcode = " << retcode << "; ";
     ret << "StringData.size = " << str.size() << "; ";
-    ret << "StringData.data = " << str << "; ";
```

Comment:
We're not printing data?

Reply:
This `string_to_hex` is only used to make a hex-dump of the data when we fail to decode it from UTF-8. So don't think it makes sense to include the non-decodable string itself. The result of this method is used as the message of an exception and we cannot pass that on to the JVM if the message itself is not valid unicode.